spyderlib/widgets/findreplace.py

```diff
@@ -191,14 +196,55 @@ def toggle_highlighting(self, state):
                 self.highlight_matches()
             else:
                 self.clear_matches()
-        
+
     def show(self):
         """Overrides Qt Method"""
         QWidget.show(self)
         self.visibility_changed.emit(True)
         if self.editor is not None:
             text = self.editor.get_selected_text()
-            if len(text) > 0:
+
+            # If no text is highlighted for search, use whatever word is under the cursor
+            if not text:
+                try:
+                    # Get line text and cursor position
+                    line_number, index = self.editor.get_cursor_line_column()
+                    block = self.editor.document().findBlockByNumber(line_number)
+                    data = to_text_string(block.text())
+                    if index < len(data):

```

Comment:
The test here boils down to "is the cursor in the line". It seems unecessary, why would it be outside ?